src/include/duckdb/parser/constraints/foreign_key_constraint.hpp

```diff
@@ -0,0 +1,46 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/parser/constraints/foreign_key_constraint.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/parser/constraint.hpp"
+#include "duckdb/common/vector.hpp"
+
+namespace duckdb {
+
+class ForeignKeyConstraint : public Constraint {
+public:
+	DUCKDB_API ForeignKeyConstraint(string pk_table, vector<string> pk_columns, vector<idx_t> pk_keys,
+	                                vector<string> fk_columns, vector<idx_t> fk_keys, bool is_fk_table);
+
+	//! The referenced table's name
+	string pk_table;
+	//! The set of main key table's columns
+	vector<string> pk_columns;
+	//! The set of main key table's column's index
+	vector<idx_t> pk_keys;
```

Comment:
Yes, this value(pk_keys) not valid at initial, but it is useful when we use alter operation after the table has been created.
When we use alter operation, we can't resolve this value from pk_columns, because catalog set is already locked, so we can't get entry of main key table.
So I've left this value in ForeignKeyConstraint, this value is empty at initial.

Reply:
I'm not a big fan of fields that are empty initially, and later filled in. This causes bugs quite easily when you use a class expecting the values to exist only to find out that they do not exist. In the Alter, do you not have access to the `BoundForeignKeyConstraint` which should have this information?